apps/aecore/lib/aecore/chain/worker.ex

```diff
@@ -86,28 +86,69 @@ defmodule Aecore.Chain.Worker do
     GenServer.call(__MODULE__, :top_height)
   end
 
+  @spec get_header_by_base58_hash(String.t()) :: Header.t() | {:error, atom()}
+  def get_header_by_base58_hash(hash) do
+    try do
+      decoded_hash = Header.base58c_decode(hash)
+      get_header(decoded_hash)
+    rescue
```

Comment:
What are you rescuing here? I did't see a throw or raise from the above funcs

Reply:
the base58c functions don't have different return values, I need to check if it was decoded correctly, if it fails then it returns invalid_hash

Reply:
I only see this crashing if the hash is <= 4bytes so the pattern match in the func will fail, but this could be prevented if we have a guard on the function with the correct size for the hash, `when byte_size(hash) == 32`